tests/jest-interactor/click-callback/test.js

```diff
@@ -0,0 +1,22 @@
+// highlight{7,13,18-21}
+import React from 'react';
+import { Button } from 'shared/components/Button';
+import Interactor from '@bigtest/interactor';
+import { mount } from '@bigtest/react';
+
+let button = new Interactor('button');
+
+it('calls "onClick" prop on button click', async () => {
+  // Render new instance in every test to prevent leaking state
+  const onClick = jest.fn();
+  // Mount the component in the DOM
+  await mount(() => <Button onClick={onClick} />);
+
+  // Use interactor `when` to assert the change has happened
+  // this will loop until `true` & fails if it isn't true by
+  // the timeout (2s by default)
+  button.when(() => {
+    expect(button.text).toEqual('Click me nao');
```

Comment:
No need to assert the button's text, as it's hardcoded. Its purpose was instead to _find_ the button using its text. Can you do this with interactor? Simply querying by `button` tag is unrealistic as in a real life app you'll have many buttons on a page.

Reply:
We currently don't have the ability to select by text on the page, but it's absolutely something we want to add (and will do so in the near-ish future). I think finding by text is _more_ fragile. In these super isolated tests it's no issue. But in acceptance tests we can't select by text because most of the data is randomized for each assertion (with faker). 😊 So our approach has been to use test selectors (`data-test-...`) that get stripped out in production builds (and some apps only appear in tests). The selectors kind of fade away and you use interactors to define how to interact with the different components on the page.

We component test in isolation with interactors (like these examples), use those same interactors in component integration tests (many components put together to build something larger), and acceptance tests. [Here's an example of a datepicker interactor](https://github.com/folio-org/ui-eholdings/blob/master/test/bigtest/interactors/datepicker.js), and then [here's how it's composed into the greater page object](https://github.com/folio-org/ui-eholdings/blob/master/test/bigtest/interactors/resource-edit.js#L111-L112) for multiple date pickers on the page. ([Here's how it's used in tests](https://github.com/folio-org/ui-eholdings/blob/master/test/bigtest/tests/custom-resource-edit-custom-coverage-test.js#L88-L112)) 

So that's the reason why we don't have the ability yet -- we don't usually select from the DOM that way. There have been many instances where we could have used it! But we ended up working around it. Anyways, that's my long-winded explanation for why we don't have it but also do want to add it :p

Reply:
Thanks for the long-winded explanation :)

>  I think finding by text is more fragile. In these super isolated tests it's no issue. 

I believe that finding by text is closer to how a person uses an app/component (they don't see the test selectors yet they can still use the feature... 🤔), and I find that expressing tests similarly to how users behave provides more value in the long term. 

But this is a philosophical debate and I know not everybody agrees. This is mentioned in the 3rd paragraph of the [about page](https://react-testing-examples.com/about/). But since you don't have a way to search by text with Interactor I'll still accept this PR because the library looks otherwise promising.

One change though, can we remote `expect(button.text).toEqual('Click me nao');` and leave just the _onClick_ assertion? I find that asserting on hardcoded values is a overkill. Similar to writing tests for constant values (which I've seen people do btw!).

Reply:
Heh, yeah, I agree. We're going to eventually implement it but currently we're working on "Validations" first. Basically we're going to assert when you say something is clickable it actually is for a11y assistive tech. 

And yeah totally! I pushed a change for that. We don't even need to hold a reference to the interactor since all we're doing is clicking.